ga4gh/server/backend.py

```diff
@@ -224,22 +235,40 @@ def logicHandler(self, logic, responses, dataset_id):
                 for id in results_arr[0]:
                     if all(id in results_arr[x] for x in range(1, len(results_arr))):
                         patient_array.append(id)
-
             return patient_array
 
         elif logic_key == 'id':
             id_list = []
-            for response in responses[logic[logic_key]]:
-                patient_id = self.getResponsePatientId(response, dataset_id)
-                if patient_id not in id_list:
-                    id_list.append(patient_id)
 
-            return id_list
+            try:
+                for response in responses[logic[logic_key]]:
+                    patient_id = self.getResponsePatientId(response, dataset_id)
+                    if patient_id not in id_list:
+                        id_list.append(patient_id)
+                if logic_negate:
+                    id_list_all = self.getAllPatientId(dataset_id, access_map)
+                    id_set = set(id_list_all) - set(id_list)
+                    id_list = list(id_set)
+                return id_list
+
+            except KeyError:
+                raise exceptions.InvalidLogicException("given id does not match a component")
```

Comment:
Maybe we can capitalize the first letter to be consistent with other exception messages we return.